### DISCUSSION TITLE
Accessing a deep property of a store inside createMemo, which point is interpreted as a dependency?

### DISCUSSION THREAD

**Discussion by aSugarRush** [solidjs/solid#1628] [Q&A] [answered]
I'm curious about how the store's proxy works inside a memo.
Suppose we have:

```js
const [store, setStore] = createStore({
  user: {
    name: 'John',
    address: {
      street: '123 Main St',
      city: 'Anytown',
      state: 'CA',
      zip: '12345'
    }
  }
});

const someMemo = createMemo(() => {
   // I'm interested in using the zip code of the address.
   const zip = store.user.address.zip
   // some other logic to justify using a memo.
   // ...
})
```

Will it declare the whole `store` as a dependency of the memo in the reactive network, or will it be the `user` property? 
Sorry if this question sounds trivial, the truth is that I have not found answers on the internet.
Thanks in advance.

**Comment by ryansolid** [comment, marked as answer]
Reactivity occurs at the getter (ie where you access the property). The store itself cannot be tracked(not 100% true but good as a mental model). So it will subscribe to `store.user`, `user.address`, `address.zip`. So in your phrasing it subscribes to the `user` property.
